projects/media-viewer/src/lib/media-viewer/viewers/image-viewer/image-viewer.component.ts

```diff
@@ -35,17 +36,17 @@ export class ImageViewerComponent {
   set zoomOperation(operation: ZoomOperation | null) {
     if (operation && !isNaN(operation.zoomFactor)) {
       this.zoom = this.updateZoomValue(+operation.zoomFactor);
-      this.zoomValue.next({ value: this.zoom });
-      this.setImageStyles();
+      this.setZoomValue(this.zoom)
+        .then(() => this.setImageStyles());
     }
   }
 
   @Input()
   set stepZoomOperation(operation: StepZoomOperation | null) {
     if (operation && !isNaN(operation.zoomFactor)) {
       this.zoom = this.updateZoomValue(this.zoom, operation.zoomFactor);
-      this.zoomValue.next({ value: this.zoom });
-      this.setImageStyles();
+      this.setZoomValue(this.zoom)
```

Comment:
can you use async await
await this.setZoomValue(this.zoom);
this.setImageStyles()